src/generators/utils/pagingOperations.ts

```diff
@@ -0,0 +1,378 @@
+          importedModels,
+          operationGroupClass
+        ).baseMethodParameters.map(parameter => {
+          if (parameter.name === "nextLink") {
+            return { ...parameter, hasQuestionToken: true };
+          }
+
+          return parameter;
+        });
+      }
+
+      const {
+        baseMethodParameters: initialMethodParameters
+      } = getOperationParameterSignatures(
+        operation,
+        clientDetails.parameters,
+        importedModels,
+        operationGroupClass
+      );
+
+      // Build an object with all the information about the paging methods
+      // while generating each of the paging methods, this will help up access
+      // information about the other methdos.
+      const pagingMethodSettings: PagingMethodSettings = {
+        initialMethod: {
+          name: `${operation.namePrefix}${initialOperationName}`,
+          parameters: initialMethodParameters
+        },
+        nextMethod: nextMethodParameters
+          ? {
+              name: `${operation.namePrefix}${nextOperationName}`,
+              parameters: nextMethodParameters
+            }
+          : undefined,
+        publicMethod: {
+          name: getPublicMethodName(initialOperationName),
+          parameters: initialMethodParameters
+        },
+        allMethod: {
+          name: `${initialOperationName}PagingAll`,
+          parameters: initialMethodParameters
+        },
+        pageMethod: {
+          name: `${initialOperationName}PagingPage`,
+          parameters: initialMethodParameters
+        },
+        bodyResponseType: bodyResponseTypeName,
+        nextLinkName: operation.pagination?.nextLinkName || "nextLink",
+        itemName: operation.pagination?.itemName || "value"
+      };
+
+      writePublicMethod(operation, operationGroupClass, pagingMethodSettings);
+      writePageMethod(operation, operationGroupClass, pagingMethodSettings);
+      writeAllMethod(operationGroupClass, pagingMethodSettings);
+    });
+}
+
+/**
+ * This method enforces Azure SDK Typescript guideline that paging methods should be named list*
+ * https://azure.github.io/azure-sdk/typescript_design.html#ts-pagination-provide-list
+ */
+function getPublicMethodName(initialOperationName: string) {
+  if (initialOperationName.indexOf("list") === 0) {
+    return initialOperationName;
+  }
+
+  if (initialOperationName.indexOf("get") === 0) {
+    return initialOperationName.replace("get", "list");
+  }
+
+  const initialName = normalizeName(initialOperationName, NameType.Class);
+  return `list${initialName}`;
+}
+
+/**
+ * Generates the content of the public method, here we reference the other 2 methods All and Page
+ */
+function writePublicMethod(
+  operation: OperationDetails,
+  operationGroupClass: ClassDeclaration,
+  pagingMethodSettings: PagingMethodSettings
+) {
+  const returnType = `PagedAsyncIterableIterator<${pagingMethodSettings.bodyResponseType}, ${pagingMethodSettings.bodyResponseType}[]>`;
```

Comment:
I think the default type for the second type argument is already the array version of the first, so perhaps we can be less verbose here:
```suggestion
  const returnType = `PagedAsyncIterableIterator<${pagingMethodSettings.bodyResponseType}>`;
```

Reply:
Yeah, good suggestion